### PR TITLE
trim style/category prefixes, add support for lora weights and lora versions

### PR DESCRIPTION
**Pull request by Efreak**
# Add support for lora versions, lora weights, and clip skip; ignore style/category prefixes for copied commands

## Detailed Description
1. add support for clip_skip parameter and in styles
2. add support for specific versions of loras by prefixing them with a "v". Ideally this would use a different parameter, but the number of parameters is running low (and there's other unimplemented features like controlnet that might eventually use them), and this is advanced usage. prefixes can be ignored by the user calling them if desired, but autocomplete was also modified to display the name of the lora just in case.
3. Allow lora clip and model strengths to be set. Implemented for styles using the same schema as the API. implemented for the advanced generate command using the format `id:modelweight:clipweight` (see above explanation regarding why it's not split into parameters)
4. Remove Style: and Category: from style names and trim spaces before processing

## Why this is needed
1. Clip Skip is wanted for some styles. [Clip skip](https://medium.com/@novita.ai/understanding-clip-skip-stable-diffusion-417b45f20645) affects how stable diffusion processes the text input, and can sort of make prompts more generic in a way (less specific).
2. civitai hosts multiple versions of loras, and horde now has support for these
3. lora clip and model weights affect how strongly the Lora affects the resulting generation
4. the style: and category: prefixes are incredibly annoying when copy/pasting commands. I have a habit of simply removing them as they're unnecessary, but I occasionally see them tripping other people up when they paste a command (once you copy/paste the command, discord loses the difference between the text you see and the internal value it has). Actually, filtering this out for model name would be nice as well, but isn't usually as much of an issue.

Overall, these changes are mostly intended to expand support for styles, but changes were made to the generate commands to support the same functions there as well.

-----

I've tested everything I changed on my own instance of the bot, but I can't help feeling like there might be something I've missed, simply because I fixed so many mistakes I made while doing this.

### CODE REVIEW

File: src/commands/party.ts [hunk 1]
```diff
@@ -159,9 +159,9 @@ export default class extends Command {
                 const styles = Object.keys(context.client.horde_styles)
                 const categories = Object.keys(context.client.horde_style_categories)
                 const available = [...styles.map(s => ({name: `Style: ${s}`, value: s})), ...categories.map(s => ({name: `Category: ${s}`, value: s}))]
-                const ret = option.value ? available.filter(s => s.value.toLowerCase().includes(option.value.toLowerCase())) : available
+                const ret = option.value ? available.filter(s => s.name.toLowerCase().includes(option.value.toLowerCase().replace("style: ","").replace("category: ","").trim())) : available
```
Comment (ZeldaFan0225) [inline src/commands/party.ts:162]: Why was this changed?
Why would you filter by the name and then replace "style" (which is case sensitive btw so that wouldnt't even work as you intended) where "value" already has the value you are calculating

Comment (Efreak) [reply src/commands/party.ts:162]: @ZeldaFan0225

> Case sensitive...

The case is already converted to lowercase. If it's easier to read, I can move the replace() before the conversation and use the capital.

> Value already had the value you are calculating

The intention is that when I do either of the following actions, the command works as intended, despite Discord losing the intended value. Currently the intended/internal value of `albedo` for the displayed text `Style: albedo` gets lost:
- copy a line that says `/party name: blah style:Style: albedo` (obtained by using the drop-down list in discord) then paste it back into discord.
- close discord with a command entered but not submitted

This applies equally to autocomplete as well as simple text entry because
1. Not having the autocomplete entry makes it seem like it will fail
2. User changing the entry by deleting a few characters and selecting again: discord loses the internal value as soon as you modify the displayed value, so you need to adjust for that if you want the autocomplete list to pop up as they edit the text.

The space is trimmed because mobile keyboards are usually configured to automatically add spaces after punctuation, and nobody's going to turn that off just for commands with one discord bot. I regularly type out style: instead of using the subcommand list; swipe text input for a single dictionary word is faster than scrolling to find the subcommand. I then use the autocomplete entry to select the style because if I type it out autocorrect will add spaces or change the word (there's so many ways this could be easier: Google to allow changing default keyboard based on app, Google to allow *the user* to disable autocorrect for a single app, Discord to disable autocorrect and grammar for command entry like terminal apps do)


These reasons apply to generate (and advanced generate) as well as parties, and should probably also be applied to model names, but it's *advanced* generate and can deal with losing some info.
I see people get confused by the style: being pasted in occasionally. I've only ever seen one complaint about the autocomplete, but it annoys me regularly.
